src/language/dafnyInstallation.ts

```diff
@@ -123,6 +129,32 @@ export class DafnyInstaller {
     }
   }
 
+  private async installFromSource() {
+    const { execSync } = require('child_process');
+    const execLog = (command: string) => {
+      this.writeStatus(`Executing: ${command}\n`);
+      execSync(command);
+    };
+    const installationPath = this.getCustomInstallationPath(os.arch());
+    await mkdirAsync(installationPath.fsPath, { recursive: true });
+    this.writeStatus(`Installing Dafny from source in ${installationPath.fsPath}.\n`);
+    const process = require('process');
+    process.chdir(installationPath.fsPath);
```

Comment:
Isn't it possible to avoid changing the working directory, i.e., adapt the commands accordingly? Then you no longer need the `process` module as well as do not mess with VSCodes working directory. If not, you should at least switch back to the original working directory. Otherwise, this could lead to undesired bugs that would probably hard for us to reproduce.